subprojects/gradle/kotlin-dsl-support/src/main/kotlin/com/avito/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -5,19 +5,15 @@ import kotlin.reflect.KProperty
 
 fun Project.getOptionalStringProperty(name: String): String? {
     return if (hasProperty(name)) {
-        property(name)?.toString()
+        val string = property(name)?.toString()
+        if (string.isNullOrBlank()) null else string
     } else {
         null
     }
 }
 
-fun Project.getOptionalStringProperty(name: String, default: String): String {
-    return if (hasProperty(name)) {
-        property(name)?.toString() ?: default
-    } else {
-        default
-    }
-}
+fun Project.getOptionalStringProperty(name: String, default: String): String =
```

Comment:
can we have one function with default value for default parameter

Reply:
nullable with default? sure. but it would be a breaking change.

Reply:
Nope, we can't actually

We can't express not-null result for this function, and it seems have a great value